### DISCUSSION TITLE
Wan 2.2 Lora Settings

### DISCUSSION THREAD

**Discussion by RandomKaos999** [kohya-ss/musubi-tuner#438] [General]
Hi all,

I'm having issues training a lora with Wan 2.2 14B FP16. It's multiple datasets, 1 with 23 images, another with 40, another with over a 100 all different classes. 1 repeat per image in all configs. Likeness is perhaps only 20% ish after 200 epochs with a 0.0002 LR on the 23 images set for example. Ive tried 0.001 and it basically acted the same, got the same level of likeness but quicker, within say 15 epochs i think it was. I let it continue up to 80 but there was no further progress made. I've tried a couple longer runs 0.00005 for example, same thing. I've tried captioning a few different ways as well, and ive got to the point I need to ask for help (takes a while to do training runs on a 3090).

This is the general jist of my command line that im using via a windows batch file:

set vae=G:\GenAI\Models\WanTraining\wan_2.1_vae.safetensors
set t5=G:\GenAI\Models\WanTraining\models_t5_umt5-xxl-enc-bf16.pth
set dit_low=G:\GenAI\Models\unet\Video\wan2.2_t2v_low_noise_14B_fp16.safetensors
set dit_high=G:\GenAI\Models\unet\Video\wan2.2_t2v_high_noise_14B_fp16.safetensors

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 "G:\GenAI\Software-New\musubi-tuner\src\musubi_tuner\wan_train_network.py" --task t2v-A14B --dit %dit_low%  --dataset_config "config\dataset.toml" --sdpa --mixed_precision fp16 --optimizer_type adamw8bit --learning_rate %learning_rate% --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module networks.lora_wan --network_dim 32 --timestep_sampling sigmoid --discrete_flow_shift 12.0 --max_train_epochs %max_train_epochs% --save_every_n_epochs 1 --seed 42 --output_dir orig_models --output_name %output_name% --logging_dir=logs --save_state --save_every_n_epochs 1 --save_last_n_epochs_state 5 --offload_inactive_dit --fp8_base --dit_high_noise %dit_high%

From my understanding, this should be training both models at the same time, producing a single lora file for each epoch, which then gets attached to both high and low on a comfy workflow and inference is as normal, ie 10 steps high, 10 steps low. Am I missing something? are my settings "correct" and i havent just found a viable learning rate yet? Anyone got any insight? 

The command line above was effectively ripped from my WAN 2.1 batch files which had success - with the required changes (as far as I can see) for WAN 2.2 to train. It does train.. it just doesnt train well. Can someone with a fresh set of eyes go over this or perhaps provide working settings that youve used so i can verify that im not insane please?

Just to be clear, I the current datasets im trying to train with, ive had success with on SDXL / Flux / Hunyuan video / WAN 2.1, I havent made any changes to them so i think its my settings somewhere, maybe im just ending it early. 

Edit: Closing this, i got a little bit drunk after posting this, bashed in some random settings and re-did the captions another different way before I crashed for the night and left it overnight. This time I managed to produce an actual working lora with a likeness much much higher (say 80%). I'll keep tinkering.
